taco-team-build.js

```diff
@@ -109,6 +109,71 @@ function applyExecutionBitFix(platforms) {
     return exec(script);
 }
 
+function checkIfXcodeAboveVersion8() {
+    var cmd = 'xcodebuild -version';
+    var execDeferred = Q.defer();
+    exec(cmd, function(err, stdout, stderr) {
+        if (err) {
+            execDeferred.reject(err);
+        } else {
+            execDeferred.resolve(stdout.toString());
+        }
+    });
+
+    return execDeferred.promise.then(function(execOutput) {
+        // parse the output here
+        var xcodeVersionRegex = /^Xcode (\d+(\.\d+)?)/;
+        var match = execOutput.match(xcodeVersionRegex);
+        if (match && match[1]) {
+            if (parseInt(match[1], 10) >= 8) {
+                return Q(true);
+            } else {
+                return Q(false);
+            }
+        }
+    });
+}
+
+function ensureProjectXcode8Compatibility(projectPath, platform, callArgs) {
+    // if platform is not ios, skip
+    if (platform !== 'ios') { return Q(); }
+    return checkIfXcodeAboveVersion8().then(function(shouldCheck) {
+        if (shouldCheck == true) {
+            // check cordova ios version
+            var cordovaIosVersionPath = path.join(projectPath, 'platforms/ios/cordova/version');
+            if (!fs.existsSync(cordovaIosVersionPath)) {
+                console.error('Cannot find version file inside Cordova iOS folder/ Check your environment.');
+                process.exit(1);
+            } else {
+                var v = require(cordovaIosVersionPath);
```

Comment:
Rather than checking for existence then requiring, I would suggest something more like

```
try {
 var v = require(cordovaIosVersionPath);
 if (semver.lt(v.version, '4.3.0')) {
   [...]
} catch {
  console.error('Cannot determine cordova-ios version');
  process.exit(1);
}
```